File: persistence/src/test/java/io/tripled/social/client/infrastructure/InMemorySocialNetworkRepositoryTest.java
package io.tripled.social.client.infrastructure;

import io.tripled.social.client.domain.SocialNetwork;
import org.junit.Before;
import org.junit.Test;

import java.time.Clock;

import static org.hamcrest.CoreMatchers.sameInstance;
import static org.junit.Assert.assertThat;

public class InMemorySocialNetworkRepositoryTest {

  private InMemorySocialNetworkRepository repository;

  @Before
  public void setUp() {
    repository = new InMemorySocialNetworkRepository(new TestDateTimeProvider(Clock.systemDefaultZone()));
  }

  @Test
  public void alwaysReturnsSameInstance() {
    SocialNetwork socialNetwork = repository.get();

    assertThat(socialNetwork, sameInstance(repository.get()));
  }
}

File: src/test/kotlin/io/tripled/social/client/infrastructure/inmemory/InMemorySocialNetworkRepositoryTest.kt
package io.tripled.social.client.infrastructure.inmemory

import io.tripled.social.client.domain.SocialNetwork
import io.tripled.social.client.infrastructure.time.TestDateTimeProvider
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.time.Clock

internal class InMemorySocialNetworkRepositoryTest {
    private var repository: InMemorySocialNetworkRepository? = null

    @BeforeEach
    fun setUp() {
        repository = InMemorySocialNetworkRepository(TestDateTimeProvider(Clock.systemDefaultZone()))
    }

    @Test
    fun alwaysReturnsSameInstance() {
        val socialNetwork = repository!!.get()

        MatcherAssert.assertThat<SocialNetwork?>(
            socialNetwork,
            Matchers.sameInstance<SocialNetwork>(repository!!.get())
        )
    }
}